quickwit/quickwit-search/src/leaf_cache.rs

```diff
@@ -0,0 +1,288 @@
+// Copyright (C) 2023 Quickwit, Inc.
+//
+// Quickwit is offered under the AGPL v3.0 and as commercial software.
+// For commercial licensing, contact us at hello@quickwit.io.
+//
+// AGPL:
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as
+// published by the Free Software Foundation, either version 3 of the
+// License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+use quickwit_proto::{LeafSearchResponse, SearchRequest, SplitIdAndFooterOffsets};
+use quickwit_storage::MemorySizedCache;
+
+pub struct LeafSearchCache {
+    content: MemorySizedCache<CacheKey, LeafSearchResponse>,
```

Comment:
We would be better off storing `LeafSearchResponse` in serialized form.
(serialized_len is a bad approximation of the amount of memory held by
LeafSearchResponse).

If the code forces you to deserialize/serialize instead of `serialize` this is ok.